addons/mail/models/mail_activity.py

```diff
@@ -375,9 +375,10 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
             right = DocumentModel.check_access_rights(doc_operation, raise_exception=False)
             if right:
                 valid_docs = DocumentModel.browse(doc_ids)._filter_access_rules(doc_operation)
-                allowed_ids += [
+                valid_doc_ids = set(valid_docs.ids)
+                allowed_ids.update({
```

Comment:
Directly use the generator
```suggestion
                allowed_ids.update(
```